flatliners/alertcorr.py

```diff
@@ -0,0 +1,122 @@
+from .baseflatliner import BaseFlatliner
+import pandas as pd
+from datetime import datetime
+
+
+class AlertCorrelation(BaseFlatliner):
+    '''
+    This class contains the code for correlation of alerts
+
+    '''
+
+    def __init__(self):
+        print(" Inside alert corr")
+        super().__init__()
+        ## Hold the values for different clusters
+        self.clusters = dict()
+        ### Empty dataframe placeholder
+        self.df = pd.DataFrame(columns=['_id', 'timestamp', 'alertname', 'gitversion'])
+        ### To keep track of previous publish
+        self.timestamp = 0
+
+    def on_next(self, x):
+        """ On each data loading it will create dataframe and then pivot frame
+        """
+        # print(f"got {x}")
+
+        if not self.metric_name(x) == 'alerts':
+            return
+
+        alert_name = self.metric_label(x, 'alertname')
+        cluster_id = self.cluster_id(x)
+        git_version = self.get_version(x)
+
+        if cluster_id not in self.clusters:
+            self.clusters[cluster_id] = dict()
+
+        if alert_name not in self.clusters[cluster_id]:
+            self.clusters[cluster_id][alert_name] = self.metric_values(x)
+
+        else:
+            previous = self.clusters[cluster_id][alert_name]
+            previous.extend(self.metric_values(x))
+            self.clusters[cluster_id][alert_name] = previous
+        previous_df = self.df
```

Comment:
no need to assign this to a variable if your just modifying it.
Just use `self.df`